clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1166,6 +1166,7 @@ public boolean shouldBlock() {
      *             function is called
      * @throws org.apache.kafka.common.errors.InterruptException if the calling thread is interrupted before or while
      *             this function is called
+     * @throws org.apache.kafka.common.errors.AuthenticationException if caller's authentication fails
```

Comment:
Nit: maybe "if authentication failed, see AuthenticationException for more details". And the javadoc for AuthenticationException can include a bit more information of the reasons, etc.